client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/util/TimeUtil.java

```diff
@@ -45,4 +46,13 @@ public static Date parseTimeField(XContentParser parser, String fieldName) throw
             "unexpected token [" + parser.currentToken() + "] for [" + fieldName + "]");
     }
 
+    public static Instant parseTimeFieldToInstant(XContentParser parser, String fieldName) throws IOException {
```

Comment:
Could `org.elasticsearch.client.transform.transforms.util.TimeUtil` class be moved to `org.elasticsearch.client.common` and reused here?